File: ambeth/Ambeth.Util/ambeth/collections/ListElem.cs
using System;

namespace De.Osthus.Ambeth.Collections
{
    public class ListElem<V> : IListElem<V>
    {
        public Object ListHandle { get; set; }

        public IListElem<V> Prev { get; set; }

        public IListElem<V> Next { get; set; }

        public V ElemValue { get; set; }

        public ListElem(V value)
        {
            this.ElemValue = value;
        }

        public void Init(V value)
        {
            this.ElemValue = value;
        }
    }
}

File: jambeth/jambeth-util/src/main/java/de/osthus/ambeth/collections/ListElem.java
package de.osthus.ambeth.collections;

public class ListElem<V> implements IListElem<V>
{
	public ListElem<V> prev = null, next = null;

	protected Object listHandle;

	public V value;

	public ListElem()
	{
	}

	public ListElem(final V value)
	{
		this.value = value;
	}

	public void init(final V value)
	{
		this.value = value;
	}

	@Override
	public Object getListHandle()
	{
		return listHandle;
	}

	@Override
	public void setListHandle(Object listHandle)
	{
		if (this.listHandle != null && listHandle != null)
		{
			throw new UnsupportedOperationException();
		}
		this.listHandle = listHandle;
	}

	@Override
	public IListElem<V> getNext()
	{
		return next;
	}

	@Override
	public void setNext(IListElem<V> next)
	{
		this.next = (ListElem<V>) next;
	}

	@Override
	public IListElem<V> getPrev()
	{
		return next;
	}

	@Override
	public void setPrev(IListElem<V> prev)
	{
		this.prev = (ListElem<V>) prev;
	}

	@Override
	public V getElemValue()
	{
		return value;
	}

	@Override
	public void setElemValue(V value)
	{
		this.value = value;
	}
}